### PR TITLE
Changed send_message_slack to zg_slack

### PR DESCRIPTION
**Pull request by marjory-ziquid**

### CODE REVIEW

File: sites/all/modules/ziquid_game/includes/functions.inc [hunk 1]
```diff
@@ -1692,5 +1691,5 @@ function zg_slack($type, $message) {
       break;
   }
 
-  return slack_send_message($slack_message, $slack_channel);
+  return zg_slack($type, $message);
```
Comment (jcheek) [inline sites/all/modules/ziquid_game/includes/functions.inc]: no.  you can't have the function call itself.

File: sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php [hunk 2]
```diff
@@ -459,18 +459,17 @@
               zg_competency_gain($game_user, 'second-mile saint');
             }
             else {
-              slack_send_message('could not give 2nd-round eq bonus for quest ' . $game_quest->id .
-                " due to $eq_success, $eq_reason, $eq_details",
-                $slack_channel);
-              $response = slack_send_message($game_user, $slack_channel);
+              zg_slack(type: 'error', message: 'could not give 2nd-round eq bonus for quest ' . $game_quest->id .
+                " due to $eq_success, $eq_reason, $eq_details");
+              $response = zg_slack($type, $message);
```
Comment (jcheek) [inline sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php]: you messed up the parameters here.

File: sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php [hunk 3]
```diff
@@ -459,18 +459,17 @@
...
               if ($response !== TRUE) {
                 firep($response, 'slack response');
               }
-              $response = slack_send_message($game_quest, $slack_channel);
+              $response = zg_slack($type, $message);
```
Comment (jcheek) [inline sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php]: you messed up the parameters here.

File: sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php [hunk 4]
```diff
@@ -459,18 +459,17 @@
...
               if ($response !== TRUE) {
                 firep($response, 'slack response');
               }
-              $response = slack_send_message($game_equipment, $slack_channel);
+              $response = zg_slack($type, $message);
```
Comment (jcheek) [inline sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php]: You messed up the parameters here.

File: sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php [hunk 5]
```diff
@@ -500,18 +499,18 @@
               zg_competency_gain($game_user, 'second-mile saint');
             }
             else {
-              slack_send_message('could not give 2nd-round st bonus for quest ' . $game_quest->id .
+              zg_slack(type: 'error', message: 'could not give 2nd-round eq bonus for quest ' . $game_quest->id .
```
Comment (jcheek) [inline sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php]: You messed up the parameters and the text here.

File: sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php [hunk 6]
```diff
@@ -500,18 +499,18 @@
...
                 " due to $st_success, $st_reason, $st_details",
                 $slack_channel);
-              $response = slack_send_message($game_user, $slack_channel);
+              $response = zg_slack($type, $message);
```
Comment (jcheek) [inline sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php]: you messed up the parameters here.

File: sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php [hunk 7]
```diff
@@ -500,18 +499,18 @@
...
               if ($response !== TRUE) {
                 firep($response, 'slack response');
               }
-              $response = slack_send_message($game_quest, $slack_channel);
+              $response = zg_slack($type, $message));
```
Comment (jcheek) [inline sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php]: you messed up the parameters here.

File: sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php [hunk 8]
```diff
@@ -500,18 +499,18 @@
...
               if ($response !== TRUE) {
                 firep($response, 'slack response');
               }
-              $response = slack_send_message($game_staff, $slack_channel);
+              $response = zg_slack($type, $message);
```
Comment (jcheek) [inline sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php]: you messed up the parameters here.

File: sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php (hunk 3 above)
Comment (marjory-ziquid) [reply sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php]: this has been fixed

File: sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php (hunk 4 above)
Comment (marjory-ziquid) [reply sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php]: this has been fixed

File: sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php (hunk 5 above)
Comment (marjory-ziquid) [reply sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php]: this has been fixed

File: sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php (hunk 6 above)
Comment (marjory-ziquid) [reply sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php]: this has been fixed

File: sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php (hunk 7 above)
Comment (marjory-ziquid) [reply sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php]: this has been fixed

File: sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php (hunk 8 above)
Comment (marjory-ziquid) [reply sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php]: this has been fixed

File: sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php [hunk 9]
```diff
@@ -457,18 +457,17 @@
...
-                $slack_channel);
-              $response = slack_send_message($game_user, $slack_channel);
+              zg_slack('error',  'could not give 2nd-round eq bonus for quest ' . $game_quest->id .
+                " due to $eq_success, $eq_reason, $eq_details");
+              $response = zg_slack('user object', $game_user);
               if ($response !== TRUE) {
                 firep($response, 'slack response');
               }
-              $response = slack_send_message($game_quest, $slack_channel);
+              $response = zg_slack('quest object' , $game_user);
```
Comment (jcheek) [inline sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php]: wrong object

File: sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php [hunk 10]
```diff
@@ -457,18 +457,17 @@
...
               if ($response !== TRUE) {
                 firep($response, 'slack response');
               }
-              $response = slack_send_message($game_equipment, $slack_channel);
+              $response = zg_slack('equipment object', $game_user);
```
Comment (jcheek) [inline sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php]: wrong object

File: sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php [hunk 11]
```diff
@@ -498,18 +497,18 @@
...
+              zg_slack('error', 'could not give 2nd-round eq bonus for quest ' . $game_quest->id .
                 " due to $st_success, $st_reason, $st_details",
                 $slack_channel);
-              $response = slack_send_message($game_user, $slack_channel);
+              $response = zg_slack('user object', $game_user);
               if ($response !== TRUE) {
                 firep($response, 'slack response');
               }
-              $response = slack_send_message($game_quest, $slack_channel);
+              $response = zg_slack('quest object', $game_user);
```
Comment (jcheek) [inline sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php]: wrong object

File: sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php [hunk 12]
```diff
@@ -498,18 +497,18 @@
...
               if ($response !== TRUE) {
                 firep($response, 'slack response');
               }
-              $response = slack_send_message($game_staff, $slack_channel);
+              $response = zg_slack('equipment  object', $game_user);
```
Comment (jcheek) [inline sites/all/modules/ziquid_game/templates/zg_quests_do.tpl.php]: extra space, wrong object

**Pull request merged**